src/classes/ADDR_Seasonal_TEST.cls

```diff
@@ -0,0 +1,923 @@
+            accAddrs2[i].Seasonal_End_Day__c = '28';
+            accAddrs2[i].MailingStreet__c = 'New Seasonal Street' + i;
+            accAddrs2[i].MailingCity__c = 'New Seasonal City' + i;
+        }
+        
+        // create additional Contact addresses, with seasonal information
+        List<Address__c> conAddrs2 = UTIL_UnitTestData_TEST.initTestAddr(2);
+        for (Integer i = 0; i < 2; i++) {
+            conAddrs2[i].Parent_Contact__c = contactsAddrs.contacts[i].Id;
+            conAddrs2[i].Default_Address__c = false;
+            conAddrs2[i].Seasonal_Start_Month__c = String.valueOf(system.today().month());
+            conAddrs2[i].Seasonal_Start_Day__c = String.valueOf(system.today().day());
+            conAddrs2[i].Seasonal_End_Month__c = String.valueOf(system.today().addMonths(1).month());
+            conAddrs2[i].Seasonal_End_Day__c = '28';
+            conAddrs2[i].MailingStreet__c = 'New Seasonal Street' + i;
+            conAddrs2[i].MailingCity__c = 'New Seasonal City' + i;
+        }
+        
+        List<Address__c> allNewAddrs = new List<Address__c>(accAddrs2);
+        allNewAddrs.addAll(conAddrs2);
+        
+        // Hack!  by setting our trigger handler as run,
+        // we can insert the new seasonal addresses but not have them processed.
+        // this way we can test the scheduled job!
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = true;
+        insert allNewAddrs;
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+
+        //Schedule the job just for test coverage purposes. But we still have to manually run the
+        //batch, as batch jobs run from a scheduled job don't actually run after stopTest, as described
+        //at https://salesforce.stackexchange.com/questions/36806/does-test-stoptest-ensure-a-system-schedule-database-batchable-completes-in-a
+        STG_InstallScript.scheduleJobIfNotScheduled('Seasonal Addresses Update', '0 15 0 * * ?', 'ADDR_Seasonal_SCHED');
+        // Get the information from the CronTrigger API object
+        CronTrigger ct = [SELECT Id, CronExpression, TimesTriggered, NextFireTime FROM CronTrigger WHERE CronJobDetail.Name = 'Seasonal Addresses Update'];
+        // Verify the expressions are the same
+        System.assertEquals('0 15 0 * * ?', ct.CronExpression);
+        // Verify the job has not run
+        System.assertEquals(0, ct.TimesTriggered);
+        // Verify the next time the job will run
+        System.assert(String.valueOf(ct.NextFireTime).contains('00:15:00'));
+        
+        // run the scheduled batch directly
+        ADDR_Seasonal_BATCH batch = new ADDR_Seasonal_BATCH();
+        Test.startTest(); 
+        Database.executeBatch(batch);
+        Test.stopTest();
+        
+        // verify Account address is new!
+        map<Id, Account> mapAccIdAcc = new map<Id, Account>([select Id, Name, BillingStreet, BillingCity, BillingState, 
```

Comment:
If the table in this query contains more than 50'000 records a runtime exception will be thrown. Consider either introducing a `WHERE` clause, or adding a `LIMIT` to contain the number of records returned.